File: shared/schemas/python/agent.py
"""
Agent Entity and Session Models
"""

from datetime import datetime
from enum import Enum
from typing import Any, Dict, List, Optional

from pydantic import BaseModel

from .base import BaseEntity


class AgentSessionStatus(str, Enum):
    QUEUED = "queued"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"


class AgentLogLevel(str, Enum):
    INFO = "info"
    THINKING = "thinking"
    ACTION = "action"
    ERROR = "error"


class WorkflowStatus(str, Enum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"


class Agent(BaseEntity):
    """Agent entity model"""

    agent_id: str
    agent_domain: str
    agent_name: str
    agent_description: str
    is_enabled: bool


class AgentSession(BaseModel):
    """Agent session model"""

    session_id: str
    user_id: str
    agent_name: str
    status: AgentSessionStatus
    created_at: datetime
    updated_at: datetime
    metadata: Optional[Dict[str, Any]] = None


class AgentRequest(BaseModel):
    """Agent request model"""

    session_id: str
    request: str
    context: Optional[Dict[str, Any]] = None
    user_id: str


class ThinkingLogEntry(BaseModel):
    """Thinking log entry model"""

    id: str
    session_id: str
    agent_name: str
    message: str
    timestamp: datetime
    log_level: AgentLogLevel


class AgentResponse(BaseModel):
    """Agent response model"""

    session_id: str
    response: str
    success: bool
    data: Optional[Any] = None
    error: Optional[str] = None
    thinking_logs: Optional[List[ThinkingLogEntry]] = None


class AgentProgressUpdate(BaseModel):
    """Agent progress update model"""

    session_id: str
    agent_name: str
    progress_percentage: float
    current_step: str
    estimated_completion: datetime


class MultiAgentWorkflow(BaseModel):
    """Multi-agent workflow model"""

    workflow_id: str
    agents_involved: List[str]
    current_agent: str
    workflow_status: WorkflowStatus
    data_flow: Dict[str, Any]
    created_at: datetime
    updated_at: datetime

File: shared/schemas/typescript/agent.ts
/**
 * Agent Entity and Session Types
 */

import { BaseEntity } from './base'

export interface Agent extends BaseEntity {
  agent_id: string
  agent_domain: string
  agent_name: string
  agent_description: string
  is_enabled: boolean
}

export interface AgentSession {
  session_id: string
  user_id: string
  agent_name: string
  status: 'queued' | 'processing' | 'completed' | 'failed'
  created_at: string
  updated_at: string
  metadata?: Record<string, any>
}

export interface AgentRequest {
  session_id: string
  request: string
  context?: Record<string, any>
  user_id: string
}

export interface AgentResponse {
  session_id: string
  response: string
  success: boolean
  data?: any
  error?: string
  thinking_logs?: ThinkingLogEntry[]
}

export interface ThinkingLogEntry {
  id: string
  session_id: string
  agent_name: string
  message: string
  timestamp: string
  log_level: 'info' | 'thinking' | 'action' | 'error'
}

export interface AgentProgressUpdate {
  session_id: string
  agent_name: string
  progress_percentage: number
  current_step: string
  estimated_completion: string
}

export interface MultiAgentWorkflow {
  workflow_id: string
  agents_involved: string[]
  current_agent: string
  workflow_status: 'pending' | 'running' | 'completed' | 'failed'
  data_flow: Record<string, any>
  created_at: string
  updated_at: string
} 